### PR TITLE
Added support for runtime analysis and add input to test results

### PR DESCRIPTION
**Pull request by hdngo**
For now, runtime analysis only happens when all test cases are cleared since I don't wanna make unneeded API calls with errored code for example. Let me know if you want this adjusted.

Based on testing with the TA seems like showing the input for tests is more intuitive for the users so we'll go along with that.

### CODE REVIEW

**Comment by hdngo** [comment]
Actually, I'll just include input for sample test cases only, wait for next commit.

Also another note, perhaps a pop up showing their runtime after all test cases a solved would be a way to display this.

**Review by weebao — CHANGES_REQUESTED** [review]
Use `response_format` for OpenAI `chat.completions.create` function. Check the inline code review comment for more details

File: app/services/execution/runtime_analysis.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            return None
+
+        try:
+            response = await self.client.chat.completions.create(
+                model="gpt-4o",
+                messages=[
+                    {"role": "system", "content": "You are a code analysis assistant. Analyze the user given code and ONLY provide a single time complexity rating e.g. O(n^2), O(log n), etc. Nothing else. Do not provide any other data or analysis."},
+                    {"role": "user", "content": f"Analyze this code:\n\n```python\n{code}\n```"}
+                ],
+                temperature=0
```
[the remark this answers is not in the document]
Comment (hdngo) [reply app/services/execution/runtime_analysis.py:40]: Thanks, I didn't know about that, please check the latest commit [130ba87](https://github.com/beatcode-official/beatcode-server/pull/13/commits/130ba87cecc1617b268db8a999db2a0ca4f560d9) 🙂

**Review by weebao — APPROVED** [review]
Looks good now

**Comment by weebao** [comment]
Closes #12

**Pull request merged**